vmdb/spec/requests/api/collections_spec.rb

```diff
@@ -0,0 +1,186 @@
+#
+# Rest API Collections Tests
+#
+require 'spec_helper'
+
+describe ApiController do
+
+  include Rack::Test::Methods
+
+  let(:api_user)   { @cfme[:user] }
+  let(:zone)       { FactoryGirl.create(:zone, :name => "api_zone") }
+  let(:miq_server) { FactoryGirl.create(:miq_server, :zone => zone) }
+  let(:template) do
+    FactoryGirl.create(:miq_template, :name => "template 1", :vendor => "vmware", :location => "template1.vmtx")
+  end
+
+  before(:each) do
+    init_api_spec_env
+  end
+
+  def app
+    Vmdb::Application
+  end
+
+  def test_collection_query(collection, collection_url, klass, attr = :id)
+    api_basic_authorize
+
+    run_get "#{collection_url}?expand=resources"
+
+    expect_query_result(collection, klass.count, klass.count)
+    expect_result_resources_to_include_data("resources", attr.to_s => klass.pluck(attr))
+  end
+
+  context "Collections" do
+
```

Comment:
will remove these in next PR, updating all files because of run_get changes. Thanks.